### ISSUE TITLE
Dynamically populate path parameters from example property

### ISSUE THREAD

**Issue by cwc-products** [google/oatts#1]
I am unable to dynamically generate the in path parameters for the tests using the example property.  Instead, it is populated with lorem ipsum.  (see example below)

I did notice you can get the path parameters to dynamically populate by using an enum property.  There also appears to be the option of using a custom values json file, but this defeats the purpose of it being dynamically generated directly from the swagger definition file.

> Example input

```
   /api/v1/avatar/generate/from/{encodedAvatar}/me.png: 
    parameters:
      - in: path
        name: encodedAvatar
        required: true
        schema:
          type: string
          example: ZXllczkvbm9zZTgvbW91dGgxL2Y2YmU1ZC9tb25zdGVy
    get:
      summary: Get the png image from the provided encoded avatar details
      tags:
        - Avatar
      responses:
        '200':
          description: OK
        '401':
          $ref: '#/components/responses/UnauthorizedError'
```

> OUTPUT

```
'use strict';
var mocha = require('mocha');
var chakram = require('chakram');
var request = chakram.request;
var expect = chakram.expect;

describe('tests for /api/v1/avatar/generate/from/{encodedAvatar}/me.png', function() {
    describe('tests for get', function() {
        it('should respond 200 for "OK"', function() {
            var response = request('get', 'http://localhost:3000/api/v1/avatar/generate/from/ut ad elit/me.png', { 
                'time': true
            });

            expect(response).to.have.status(200)
            return chakram.wait();
        });    
    });
});
```

**Comment by noahdietz** [comment]
Hey @cwc-products ! Thanks for the ticket! If I understand the ask here its to have request parameters respect the given `example` when available, and use spoofed data when an `example` is not available. Please let me know if I am not understanding.

I believe I would extend this to the current implementation to say that the priority would go to data provided via `custom-values`, then in-spec `examples`, then default to spoofed data. 

The custom value option is useful when you want to 1. generate multiple test suites with different data and 2. you don't want to clutter your API spec with potentially large examples. In the example you've provided it might seem like overkill, but I believe it is useful for data injection during generation.

If my understanding/proposal is correct, I can work on it soon, or you can submit a PR if you are interested in contributing. Let me know how you'd like to proceed.

Thanks again for the ticket.

**Comment by noahdietz** [comment]
It also looks like from that example that the spoofed path param data isn't URL safe. I will open a separate bug for that!

**Comment by cwc-products** [comment]
Hi @noahdietz, thanks for the quick response.  It looks like you understand the request correctly.  I would love to contribute and given I have time I'll get a feature branch up and running.

Thanks

**Comment by noahdietz** [comment]
Awesome! Make sure to take a look at the contributing guidelines for the repo and I look forward to your PR.

**Comment by noahdietz** [comment]
Hey @cwc-products just wanted to check in and see if you have any questions on this enhancement.

I've assigned it to you, but if at any point you don't think you'll be able to finish it up, just assign it back to me or comment here and I can pick it up! Thanks again for helping out and contributing!

**Comment by cwc-products** [comment]
@noahdietz thanks for checking in, I haven't had a chance to start work on this due to shifting priorities, but I hope to take a look at it this week.

**Comment by noahdietz** [comment]
@cwc-products hey checking in! I know that things are busy and I really appreciate your willingness to help & get involved. If you still want to take the issue, I'm happy to support that! But if at any point you'd like to relinquish the issue, that is totally cool :smile: just reassign!

**Comment by noahdietz** [comment]
The fix in #25 was released in oatts@1.4.1. Please give it a shot.
